tests/text/test_tabstops.py

```diff
@@ -34,9 +34,30 @@ def it_can_iterate_over_its_tab_stops(self, iter_fixture):
         assert count == expected_count
         assert TabStop_.call_args_list == expected_calls
 
+    def it_can_get_a_tab_stop_by_index(self, index_fixture):
+        tab_stops, expected_vals, tab_stop_, TabStop_, expected_calls = (
+            index_fixture
+        )
+        for count in range(len(tab_stops)):
+            assert tab_stops[count] is tab_stop_
+        assert TabStop_.call_args_list == expected_calls
+
```

Comment:
Btw, in response to a specific question in your prior comment:

The TabStop_ mock gets called here every time your code calls TabStop(); setting up that class mock replaces the TabStop object in the tabstops module with TabStop_. When TabStop_ gets called, it doesn't create a TabStop object, but it does remember every time it was called and with what. So when we test `TabStop_.assert_called_once_with(tab)`, we're testing to make sure our code called TabStop exactly once, and passed the w:tab element of interest as a parameter.

A mock also has a discoverable return value on .return_value. So we set TabStop_.return_value to the tab_stop_ mock (that's done in the component definition), then we can test that our code returned the return value of TabStop, as it should.